### ISSUE TITLE
Tor browser, Welcome screen - some elements not loaded

### ISSUE THREAD

**Issue by sorooris** [trezor/trezor-suite#3999]
**Describe the bug**
Elements @connect-device-prompt and @onboarding/expand-troubleshooting-tips are loaded but invisible in the Tor browser.

**Info:**
 - Remembered Wallet: N/A
 - Suite Version: web 21.6.2 [290aab68e](https://github.com/trezor/trezor-suite/commits/290aab68e418b837f9bed8a8839143dddaefb81c) 
 - Browser: Tor browser 10.0.17
 - OS: any (tested NixOS 21.05)
 - Bridge: 2.0.31
 - Firmware Version: any

**Steps to reproduce:**
1. Launch Tor browser,
2. Access web Suite,
3. Observe the Welcome screen.

**Expected result:**
All elements are loaded correctly.

**Actual result:**
@connect-device-prompt and @onboarding/expand-troubleshooting-tips elements are not displayed.

**Screenshots:**
![image](https://user-images.githubusercontent.com/10185314/124893317-e8691f00-dfda-11eb-9c02-f6c77ec2f092.png)

**Comment by jerechinsky** [comment]
@sorooris could you re-test please? I think something related has been addressed.

**Comment by sorooris** [comment]
@alex-jerechinsky 

**QA NOK**

Issue still persists.

Remembered Wallet: N/A
Suite Version: web 21.8.1 bde9e100
Browser: Tor browser 10.0.18
OS: any (tested NixOS 21.05)
Bridge: 2.0.31
Firmware Version: any

![image](https://user-images.githubusercontent.com/10185314/130614364-1cbd5828-9946-4424-a43f-ce2900e308b2.png)

**Comment by jerechinsky** [comment]
Technically the only supported browsers are Chrome and Firefox. On top of that we already have Tor functionality in Suite on desktop.

@Hermez-cz @tsusanka do you think we should try to fix this one? Or should we rule it out because Tor browser is not officially supported?

**Comment by tsusanka** [comment]
> @Hermez-cz @tsusanka do you think we should try to fix this one? Or should we rule it out because Tor browser is not officially supported?

I will leave this one up to the Product team.

**Comment by tsusanka** [comment]
I have tested this in tor-browser (10.5.8 (based on Mozilla Firefox 78.15.0esr)) and I do see the components just fine:

![image](https://user-images.githubusercontent.com/1835345/139697557-4f302226-14de-41c7-b784-dbdd8b040586.png)

I can't however connect to Bridge (2.0.31):  `Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at http://127.0.0.1:21325/. (Reason: CORS request did not succeed).` My guess is that Tor blocks this in general, but I think @prusnak mentioned this used to work before?

Also @STew790 could you please retest this with Bridge 2.0.31 on let's say Windows/Mac just to make sure this is not some cryptic setup of mine?

**Comment by prusnak** [comment]
> but I think @prusnak mentioned this used to work before?

There used to be the instructions in the Wiki, but Alex deleted them:

old revision: https://wiki.corp.sldev.cz/index.php?title=Glossary:Tor&oldid=60123

TL;DR you need to set `network.http.referer.hideOnionSource` to `true` in the Advanced settings. Otherwise pages do not report referers for onion pages.

**Comment by tsusanka** [comment]
Thanks. I vaguely remember I have ACKed this, so this is on me. We'll put it back, I probably haven't realized this a legitimite  use-case.

So I am closing this issue then, thanks.

**Comment by sunknudsen** [comment]
Trying to get https://suite.trezor.io/web/ and https://connect.trezor.io/8/popup.html#loading working on Tails/Tor Browser.

Trezor Bridge is installed and I configured Tor Browser (network.proxy.no_proxies_on 👉 127.0.0.1:21325), but I get “Steps to make sure Trezor Bridge is running”.

Using `https` vs `onion` because this is how third parties integrate Trezor (see https://adalite.io/).

```
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at http://127.0.0.1:21325/. (Reason: CORS request did not succeed).
```

Is this something we can “patch” at our end? According to Firefox [docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS/Errors/CORSDidNotSucceed), as of Firefox 68, `https` pages are not permitted to access `http://localhost`, although this may be changed by [Bug 1488740](https://bugzilla.mozilla.org/show_bug.cgi?id=1488740).

Fixing this would likely open a world of possibilities on Tails.

@prusnak, would you happen to know a way around this?

**Comment by prusnak** [comment]
> @prusnak, would you happen to know a way around this?

Did you set `network.http.referer.hideOnionSource` to `false`? Mentioned at the bottom of https://wiki.trezor.io/Glossary:Tor 

The bridge needs to "see" the referrer, otherwise the CORS does not work.

**Comment by sunknudsen** [comment]
@prusnak Thanks for helping out… no given I am using `https` version vs `http://…onion`.

My understanding is that `network.http.referer.hideOnionSource` is only required when using onion addresses.

Btw, noticed https://suite.trezor.io/web/ does not suggest onion address (“.onion available”) like I do on https://sunknudsen.com/.

Is this bad practice? Wanted to try `network.http.referer.hideOnionSource`, but having a hard time finding official onion addresses.

That said, https://adalite.io/ (for example) uses https://connect.trezor.io/8/popup.html#loading so `network.http.referer.hideOnionSource` would likely not help right?

**Comment by prusnak** [comment]
> Btw, noticed https://suite.trezor.io/web/ does not suggest onion address (“.onion available”) 

It should:

```
$ curl -q https://suite.trezor.io/web/ | grep onion-location
...<meta httpequiv="onion-location" content="http://suite.trezoriovpjcahpzkrewelclulmszwbqpzmzgub37gbcjlvluxtruqad.onion/web"/>...
```

With that onion address, the web version of the Suite should work in the Tor browser. If it doesn't work with the onion address please open a new issue.

If the ".onion available" button does not show for you, please open a new issue.

If AdaLite does not work, please open an issue in their repository.

**Comment by sunknudsen** [comment]
> If the ".onion available" button does not show for you, please open a new issue.

https://github.com/trezor/trezor-suite/issues/4626

> If AdaLite does not work, please open an issue in their repository.

I am pretty sure issue is related to https://github.com/trezor/trezor-suite/issues/3999#issuecomment-989805212 not AdaLite.

**Comment by sorooris** [comment]
@sunknudsen big fan of yours, I'll make myself available for any retests if needed, yours truly, former Trezor QA lead, just lemme know.

**Comment by sunknudsen** [comment]
Thanks for helping out @sorooris.
